torchaudio/prototype/models/conformer_wav2vec2.py

```diff
@@ -0,0 +1,298 @@
+from typing import List, Optional, Tuple, Union
+
+from torch import nn, Tensor
+from torch.nn import Module
+from torchaudio.models import Wav2Vec2Model
+from torchaudio.models.conformer import ConformerLayer
+from torchaudio.models.rnnt import _TimeReduction
+from torchaudio.models.wav2vec2 import components
+
+
+class TimeReduction(Module):
```

Comment:
would it be better to name it as `FeatureExtractor` or `ConformerFeatureExtractor`? want to hear thoughts of other folks @hwangjeff @mthrok @xiaohui-zhang

Reply:
That sounds more fitting to me as the docstring says "Extract features from input. Consists of time reduction **and** linear layer."

Reply:
yeah if we're using the docstring as the reference, `FeatureExtractor` sounds good. although, if we're saying that the output of this module is features, what's the input?

Reply:
docstring was copied over from w2v2 model when I had originally named the class ConformerFeatureExtractor, and the input is waveform transforms (logmel spectrograms), so not too sure if feature extractor is fully appropriate for the name

Reply:
renaming it to `FeatureEncoder` for now, as is used in the [paper](https://arxiv.org/pdf/2110.07313.pdf). can follow up on naming if there are any other ideas